### PR TITLE
Detect when sending email is disbled and adjust accordingly

### PR DESCRIPTION
**Pull request by mbabker**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/21646

### Summary of Changes

When sending mail is disabled on a site, we cannot process information requests.  Therefore, the UI is adjusted to:

- Show a warning on the frontend form saying that requests cannot be created
- Hide the "New" button in the admin list view
- Fail the save actions in the models when creating new requests
- Force a redirect in the admin when trying to navigate to the request view's edit layout (create a request)
- Hide the button to email the data export to users on the list and item views

### Testing Instructions

Try to create an information request when mail sending is disabled.  You should be met with warnings and an inability to create the request.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.com_privacy.ini [hunk 1]
```diff
@@ -58,6 +58,7 @@ COM_PRIVACY_EMAIL_DATA_EXPORT_COMPLETED_BODY="An administrator for [URL] has com
 COM_PRIVACY_EMAIL_DATA_EXPORT_COMPLETED_SUBJECT="Data Export for [SITENAME]"
 COM_PRIVACY_ERROR_ACTIVE_REQUEST_FOR_EMAIL="There is already an active information request for this email address, the active request should be completed before starting a new one."
 COM_PRIVACY_ERROR_CANNOT_CREATE_REQUEST_FOR_SELF="You cannot create an information request for yourself."
+COM_PRIVACY_ERROR_CANNOT_CREATE_REQUEST_WHEN_SENDMAIL_DISABLED="Information requests cannot be created when sending mail is disabled."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.com_privacy.ini]: Your choice but most of the time the contraction "can't" has been used instead of cannot

Comment (mbabker) [reply administrator/language/en-GB/en-GB.com_privacy.ini]: We should do this as a separate PR IMO then.  Clearly "can't" isn't in my writing style as all the strings for `com_privacy` use "cannot".

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_privacy.ini]: Not mine either but its in the style guide ;)

When you merge this one I will do the pr here

**Comment by alikon** [comment]
from backend with **send mail enabled**
create a new privacy request
screenshot from 2018-08-19 09-09-53

**Comment by mbabker** [comment]
Fixed.

**Pull request merged**